src/content/docs/explanations/language/programs.mdx

```diff
@@ -0,0 +1,245 @@
+allowing your browser to access it by routing the data through the platform.
+:::
+
+## Statement chaining
+
+You chain statements with either a newline (`\n`) or pipe symbol (`|`). We
+purposefully offer choice to cater to two primary styles:
+
+1. Vertical structuring with newlines for full-text editing
+2. Horizontal inline pipe composition for command-line usage
+
+Prefer the vertical approach for readability in files and documentation.
+Throughout this documentation, we only use the vertical style for clarity and
+consistency.
+
+Let's juxtapose the two styles. Here's a vertical TQL program:
+
+```tql
+let $ports = [22, 443]
+
+from_file "/tmp/logs.json"
+where port in $ports
+select src_ip, dst_ip, bytes
+summarize src_ip, total=sum(bytes)
+```
+
+And here a horziontal one:
+
+```tql
+let $ports = [22, 443] | from "/tmp/logs.json" | where port in $ports | select src_ip, dst_ip, bytes | summarize src_ip, total=sum(bytes)
+```
+
+In theory, you can combine pipes and newlines to write programs that resemble
+Kusto and similar languages. However, we discourage this practice because it can
+make the code harder to read and maintainespecially when adding nested
+pipelines that increase the level of indentation.
+
+## Diagnostics
+
+TQL's diagnostic system is designed to give you insights into what happens
+during data processing. There exist two types of diagnostics:
+
+1. **Errors**: Stop pipeline execution immediately (critical failures)
+2. **Warnings**: Signal data quality issues but continue processing
+
+When a pipeline emits an error, it stops execution. Unless you configured the
+pipeline to restart on error, it now requires human intervention to resolve the
+issue and resume execution.
+
+Warnings do not cause a screeching halt of the pipeline. They are useful for
+identifying potential issues that may impact the quality of the processed data,
+such as missing or unexpected values.
+
+:::tip[Best Practices]
+We have a dedicated [section on warnings and
+errors](/tutorials/learn-idiomatic-tql/#data-quality) in our [learning idiomatic
+TQL tutorial](/tutorials/learn-idiomatic-tql).
+:::
+
+## Pipeline nesting
+
+Operators can contain entire subpipelines that execute based on the operator's
+semantics. For example, the [`every`](/reference/operators/every) operator
+executes its subpipeline at regular intervals:
+
+```tql
+every 1h {
+  from_http "api.example.com"
+  select domain, risk
+  context::update "domains", key=domain, value=risk
+}
+```
+
+You define subpipelines syntactically within a block of curly braces (`{}`).
+
+Some operators require that you define a closed (void-to-void) pipeline, whereas
+others exhibit parsing (bytes-to-events) or printing (events-to-bytes)
+semantics.
+
+## Parallel processing
+
+The [`fork`](/reference/operators/fork) operator splits a pipeline into multiple
+parallel legs. It takes a nested pipeline as an argument that represents the
+secondary leg.
```

Comment:
All TQL pipelines are parallel. `fork` is about copying a stream of data. I would remove this from the page here cause it's just a very special use case and not relevant here.